pkg/apis/core/validation/shoot.go

```diff
@@ -1796,6 +1819,19 @@ func validateShootOperation(operation string, shoot *core.Shoot, fldPath *field.
 	return allErrs
 }
 
+func validateShootMaintenanceOperation(operation string, shoot *core.Shoot, fldPath *field.Path) field.ErrorList {
```

Comment:
Currently some of the values that can be added to the maintenance annotation `maintenance.gardener.cloud/operation` e.g. `ShootOperationRotateCAComplete` can cause an error if the corresponding first step (`ShootOperationRotateCAStart`) has not been executed yet. So they should also be validated for the maintenance annotation similar to how it is done for the shoot operation annotation.
Perhaps we can use one method for both validations and just pass the annotation values, key and valid fields to this method.

Reply:
If `gardener.cloud/operation` and `maintenance.gardener.cloud/operation` were mutually exclusive they could share the same validation. WDYT?

Reply:
Yes, if we add validation for the rotation operations in `maintenance.gardener.cloud/operation`, then we must make them mutually exclusive.

The other thing that I was thinking about, instead of adding this validation for `maintenance.gardener.cloud/operation`, to instead check in `IsValidShootOperation` if the rotation operation specified in `maintenance.gardener.cloud/operation` can be set as `gardener.cloud/operation`. If it cannot then `getOperation` here: https://github.com/gardener/gardener/blob/23610eb7d088ec52036d57a20daab8e12cb20bf0/pkg/controllermanager/controller/shoot/shoot_maintenance_control.go#L246 could simply return `reconcile`. Otherwise the shoot would not be maintained.

Reply:
@plkokanov I implemented the logic mentioned in my previous comment in the first new commit and deleted the obsolete `IsValidShootOperation` function in the second.
Do you mind having another look?